simulation/simple_sensor_simulator/include/simple_sensor_simulator/vehicle_simulation/vehicle_model/sim_model_delay_steer_map_acc_geared.hpp

```diff
@@ -0,0 +1,338 @@
+  if (base_keys.size() < 2 || base_values.size() < 2) {
+    throw std::invalid_argument(
+      "The size of points is less than 2. base_keys.size() = " + std::to_string(base_keys.size()) +
+      ", base_values.size() = " + std::to_string(base_values.size()));
+  }
+
+  // when sizes of indices and values are not same
+  if (base_keys.size() != base_values.size()) {
+    throw std::invalid_argument("The size of base_keys and base_values are not the same.");
+  }
+}
+}  // namespace interpolation_utils
+
+// copied from https://github.com/tier4/autoware.universe/blob/v0.17.0/common/interpolation/include/interpolation/linear_interpolation.hpp
+namespace interpolation
+{
+double lerp(const double src_val, const double dst_val, const double ratio);
+
+std::vector<double> lerp(
+  const std::vector<double> & base_keys, const std::vector<double> & base_values,
+  const std::vector<double> & query_keys);
+
+double lerp(
+  const std::vector<double> & base_keys, const std::vector<double> & base_values,
+  const double query_key);
+
+}  // namespace interpolation
+
+// copied from https://github.com/autowarefoundation/autoware.universe/blob/main/simulator/simple_planning_simulator/include/simple_planning_simulator/utils/csv_loader.hpp
+using Table = std::vector<std::vector<std::string>>;
+using Map = std::vector<std::vector<double>>;
+class CSVLoader
+{
+public:
+  explicit CSVLoader(const std::string & csv_path);
+
+  bool readCSV(Table & result, const char delim = ',');
+  static bool validateData(const Table & table, const std::string & csv_path);
+  static bool validateMap(const Map & map, const bool is_col_decent);
+  static Map getMap(const Table & table);
+  static std::vector<double> getRowIndex(const Table & table);
+  static std::vector<double> getColumnIndex(const Table & table);
+  static double clampValue(
+    const double val, const std::vector<double> & ranges, const std::string & name);
+
+private:
+  std::string csv_path_;
+};
+
+class AccelerationMap
+{
+public:
+  bool readAccelerationMapFromCSV(const std::string & csv_path)
+  {
+    CSVLoader csv(csv_path);
+    std::vector<std::vector<std::string>> table;
+    if (!csv.readCSV(table)) {
+      std::cerr << "[SimModelDelaySteerMapAccGeared]: failed to read acceleration map from "
+                << csv_path << std::endl;
+      return false;
+    }
+
+    vehicle_name_ = table[0][0];
+    vel_index_ = CSVLoader::getRowIndex(table);
+    acc_index_ = CSVLoader::getColumnIndex(table);
+    acceleration_map_ = CSVLoader::getMap(table);
+
+    std::cout << "[SimModelDelaySteerMapAccGeared]: success to read acceleration map from "
+              << csv_path << std::endl;
+    return true;
+  }
+
+  double getAcceleration(const double acc_des, const double vel) const
+  {
+    std::vector<double> interpolated_acc_vec;
+    const double clamped_vel = CSVLoader::clampValue(vel, vel_index_, "acc: vel");
+
+    // (throttle, vel, acc) map => (throttle, acc) map by fixing vel
```

Comment:
Please use doxygen [note command.](https://www.doxygen.nl/manual/commands.html#cmdnote)

Reply:
This is imported from simple_planning_simulator original file, so it is not desirable to fix.

Reply:
In this place, I found that the note command is invalid to `doxygen`.